lib/hammer_cli_foreman_discovery/discovery.rb

```diff
@@ -83,6 +90,21 @@ def request_params
       build_options
     end
 
+    class AutoProvisionCommand < HammerCLIForeman::SingleResourceCommand
+      action :auto_provision
+      command_name "auto_provision"
+      desc _("Auto provision a host")
+      success_message _("Host created")
+      failure_message _("Could not create the host")
+
+      def self.build_options
+        option "--id", "ID", " "
+        option "--name", "NAME", "The Name to search by"
+      end
+
+      build_options
```

Comment:
`build_options` are used only for building options automatically form an API docs. You can either rely on the docs and remove our `build_options` redefinition or call `option` directly in the command class.